### PR TITLE
Unify txpool.remove function

### PR DESCRIPTION
**Pull request by loocapro**
Related to: https://github.com/paradigmxyz/reth/issues/6617

### CODE REVIEW

File: crates/transaction-pool/src/pool/txpool.rs [hunk 1]
```diff
@@ -2818,4 +2818,29 @@ mod tests {
...
+        // Add them to the pool
+        let _res = pool.add_transaction(v0.clone(), on_chain_balance, on_chain_nonce).unwrap();
+        let _res = pool.add_transaction(v1.clone(), on_chain_balance, on_chain_nonce).unwrap();
+
+        assert_eq!(0, pool.queued_transactions().len());
+        assert_eq!(2, pool.pending_transactions().len());
+
+        // Remove first (nonce 0) - simulating that it was taken to be a part of the block.
+        pool.remove_transactions(vec![*v0.hash()]);
+    }
```
Comment (mm-zk) [inline crates/transaction-pool/src/pool/txpool.rs:2845]: is the other test, that I've mentioned in the bug report, also passing after your changes ?

File: crates/transaction-pool/src/pool/txpool.rs [hunk 2]
```diff
@@ -2818,4 +2818,29 @@ mod tests {
...
+
+        // Add them to the pool
+        let _res = pool.add_transaction(v0.clone(), on_chain_balance, on_chain_nonce).unwrap();
+        let _res = pool.add_transaction(v1.clone(), on_chain_balance, on_chain_nonce).unwrap();
+
+        assert_eq!(0, pool.queued_transactions().len());
+        assert_eq!(2, pool.pending_transactions().len());
+
+        // Remove first (nonce 0) - simulating that it was taken to be a part of the block.
+        pool.remove_transactions(vec![*v0.hash()]);
```
Comment (mm-zk) [inline crates/transaction-pool/src/pool/txpool.rs:2844]: I'd suggest adding some assert after this line to check that the second transaction is really at the top of the queue.

File: crates/transaction-pool/src/pool/txpool.rs (hunk 1 above)
Comment (loocapro) [reply crates/transaction-pool/src/pool/txpool.rs:2848]: Nope:

```rust
 left: [1, 3, 2, 3]
 right: [1, 2, 3]
```

**Review by mattsse — APPROVED** [review]
@Rjected 

this removes the distinction between remove and prune, and fixes the bug if `TransactionPool::remove_transaction` is called manually, which we currently only do in auto-seal

https://github.com/paradigmxyz/reth/blob/c2b7a386d87e8b3a82e666bef0bffcf985cc16e1/crates/transaction-pool/src/traits.rs#L270-L277

This isn't correct now, and I think we should change this to adhere to the docs?

now we're advancing the independent tx on remove 
this does not affect how we remove mined transactions
and also does not affect how we discard txs because we remove all descendant txs there

pending @Rjected

**Review by Rjected — APPROVED** [review]
lgtm, yeah the trait docs should be updated

**Pull request merged**